tests/Resources/ti.geolocation.test.js

```diff
@@ -307,12 +307,104 @@ describe.windowsBroken('Titanium.Geolocation', () => {
+			// callback never gets called on iOS sim
+			it.iosBroken('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				Ti.Geolocation.getCurrentHeading(function (data) {
+					try {
+						should(data).have.property('success').which.is.a.Boolean();
+						should(data.success).be.eql(true);
+						should(data).have.property('code').which.is.a.Number();
+						should(data.code).be.eql(0);
+						should(data.heading).be.an.Object();
+						should(data.heading).have.a.property('magneticHeading').which.is.a.Number();
+						should(data.heading).have.a.property('accuracy').which.is.a.Number();
+					} catch (err) {
+						return finish(err);
+					}
+					finish();
+				});
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				const result = Ti.Geolocation.getCurrentHeading();
+				result.should.be.a.Promise();
+				if (OS_IOS) {
+					// The promise never gets called on iOS Sim
+					return finish();
+				}
+				result.then(data => {
+					should(data).have.property('success').which.is.a.Boolean();
+					should(data.success).be.eql(true);
+					should(data).have.property('code').which.is.a.Number();
+					should(data.code).be.eql(0);
+					should(data.heading).be.an.Object();
+					should(data.heading).have.a.property('magneticHeading').which.is.a.Number();
+					should(data.heading).have.a.property('accuracy').which.is.a.Number();
+					return finish();
+				}).catch(e => finish(e));
+			});
 		});
 
 		describe('#getCurrentPosition()', () => {
 			it('is a Function', () => {
 				should(Ti.Geolocation).have.a.property('getCurrentPosition').which.is.a.Function();
 			});
+
+			// This never fires from my android emulator
+			it.androidBroken('works via callback argument', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				Ti.Geolocation.getCurrentPosition(function (data) {
+					try {
+						should(data).have.property('success').which.is.a.Boolean();
+						should(data.success).be.eql(true);
+						should(data).have.property('code').which.is.a.Number();
+						should(data.code).be.eql(0);
+						should(data.coords).be.an.Object();
+						should(data.coords).have.a.property('latitude').which.is.a.Number();
+						should(data.coords).have.a.property('longitude').which.is.a.Number();
+						should(data.coords).have.a.property('altitude').which.is.a.Number();
+						should(data.coords).have.a.property('accuracy').which.is.a.Number();
+						should(data.coords).have.a.property('heading').which.is.a.Number();
+						should(data.coords).have.a.property('speed').which.is.a.Number();
+						should(data.coords).have.a.property('timestamp').which.is.a.Number();
+					} catch (err) {
+						return finish(err);
+					}
+					finish();
+				});
+			});
+
+			it('works via Promise return value', function (finish) {
+				this.timeout(6e4); // 60 sec
+
+				const result = Ti.Geolocation.getCurrentPosition();
+				result.should.be.a.Promise();
+				if (OS_ANDROID) {
```

Comment:
<!--
  1 failure:  tests/Resources/t...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.geolocation.test.js;
  Line: 388;
-->
- :no_entry_sign: tests/Resources/ti.geolocation.test.js line 388 – 'OS_ANDROID' is not defined. (no-undef)